src/main/java/pl/allegro/tech/opel/OpelParser.java

```diff
@@ -343,14 +347,36 @@ protected Rule fromStringLiteral(String string) {
         return String(string).label("'" + string + "'");
     }
 
+    /**
+     * Escapes special characters in string literals.
+     * Explicitly converts \t and \n to their corresponding whitespace characters.
+     * All other escaped characters are left as they are.
+     */
     protected String escapeString(String string) {
+        Matcher matcher = escapeCharacterPattern.matcher(string);
         StringBuilder result = new StringBuilder();
-        for (int i = 0; i < string.length(); i++) {
-            if (string.charAt(i) == '\\') {
-                i++;
+        while (matcher.find()) {
```

Comment:
Maybe we can just have 
```
if (string.charAt(i) == '\\') {
 var nextChar = string.charAt(i+1) // check if i+1 exists
 if(nextChar == 't') ...
}
```
Not sure about regex performance of regex in this place.

Reply:
I'll look into that, when we agree what's the approach for this.
We could abandon this change all together and just leave it as is - since if we want to handle \t and \n we're changing global behaviour (not related to startsWith alone)